spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java

```diff
@@ -0,0 +1,799 @@
+            .setPathParam("owner_key", ownerKey)
+            .addQueryParam("page", "1")
+            .addQueryParam("per_page", "1")
+            .execute();
+        assertNotNull(response);
+        assertEquals(200, response.getCode());
+        assertThat(response.deserialize(new TypeReference<List<ProductDTO>>() {}))
+            .isNotNull()
+            .hasSize(1)
+            .containsOnly(prod1);
+
+        response = Request.from(client)
+            .setPath(OWNER_PRODUCTS_PATH)
+            .setPathParam("owner_key", ownerKey)
+            .addQueryParam("page", "2")
+            .addQueryParam("per_page", "1")
+            .execute();
+        assertNotNull(response);
+        assertEquals(200, response.getCode());
+        assertThat(response.deserialize(new TypeReference<List<ProductDTO>>() {}))
+            .isNotNull()
+            .hasSize(1)
+            .containsOnly(prod2);
+
+        response = Request.from(client)
+            .setPath(OWNER_PRODUCTS_PATH)
+            .setPathParam("owner_key", ownerKey)
+            .addQueryParam("page", "3")
+            .addQueryParam("per_page", "1")
+            .execute();
+        assertNotNull(response);
+        assertEquals(200, response.getCode());
+        assertThat(response.deserialize(new TypeReference<List<ProductDTO>>() {}))
+            .isNotNull()
+            .hasSize(1)
+            .containsOnly(prod3);
+
+        response = Request.from(client)
+            .setPath(OWNER_PRODUCTS_PATH)
+            .setPathParam("owner_key", ownerKey)
+            .addQueryParam("page", "4")
+            .addQueryParam("per_page", "1")
+            .execute();
+        assertNotNull(response);
+        assertEquals(200, response.getCode());
+        assertThat(response.deserialize(new TypeReference<List<ProductDTO>>() {}))
+            .isNotNull()
+            .hasSize(0);
+    }
+
+    @Test
+    @DisplayName("should list products in sorted pages")
+    public void shouldListProductsInSortedPages() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        String ownerKey = owner.getKey();
+        // The creation order here is important so we don't accidentally setup the correct ordering by
+        // default.
+        ProductDTO prod2 = ownerProductApi.createProductByOwner(ownerKey, Products.random()
+            .id("test_product-2"));
+        sleep(1000);
+        ProductDTO prod1 = ownerProductApi.createProductByOwner(ownerKey, Products.random()
+            .id("test_product-1"));
+        sleep(1000);
+        ProductDTO prod3 = ownerProductApi.createProductByOwner(ownerKey, Products.random()
+            .id("test_product-3"));
+
+        Response response = Request.from(client)
+            .setPath(OWNER_PRODUCTS_PATH)
+            .setPathParam("owner_key", ownerKey)
+            .addQueryParam("page", "1")
+            .addQueryParam("per_page", "1")
+            .addQueryParam("sort_by", "id")
```

Comment:
We should specify the sort order here so we're not at the risk of the default changing breaking this test. Otherwise the test logic needs to be updated to be handle either order, or the test name needs to be updated to indicate the default ordering is part of the test.

Reply:
I see! Updated to ascending sort order